examples/structured_data/advanced_techniques_for_structured_data.py

```diff
@@ -0,0 +1,472 @@
+            # Create a numeric feature column.
+            feature_column = tf.feature_column.numeric_column(feature_name)
+        else:
+            vocabulary = CATEGORICAL_FEATURES_WITH_VOCABULARY[feature_name]
+            # Create a categorical feature column.
+            feature_column = tf.feature_column.categorical_column_with_vocabulary_list(
+                feature_name, vocabulary_list=vocabulary
+            )
+            # Set the embedding dimensions to the square root of the vocabulary size.
+            embedding_dims = int(math.sqrt(len(vocabulary)))
+            # Encode the categorical feature with embedding representation.
+            feature_column = tf.feature_column.embedding_column(
+                feature_column, dimension=embedding_dims
+            )
+        feature_columns.append(feature_column)
+
+    # Create a crossed feature to capture the interaction between Soil_Type and Wilderness_Area.
+    num_values = int(
+        len(CATEGORICAL_FEATURES_WITH_VOCABULARY["Soil_Type"])
+        * len(CATEGORICAL_FEATURES_WITH_VOCABULARY["Wilderness_Area"])
+    )
+    soil_type_X_wilderness_area = tf.feature_column.crossed_column(
+        ["Soil_Type", "Wilderness_Area"], hash_bucket_size=num_values
+    )
+    # Encode the crossed feature with embedding representation.
+    soil_type_X_wilderness_area_embedding = tf.feature_column.embedding_column(
+        soil_type_X_wilderness_area, dimension=int(math.sqrt(num_values))
+    )
+    feature_columns.append(soil_type_X_wilderness_area_embedding)
+
+    return feature_columns
+
+
+"""
+## Experiment 1: baseline model
+
+In first experiment, we create a multi-layer feed-forward network, where the categorical features are
+one-hot encoded.
+"""
+
+
+def create_baseline_model():
+    inputs = create_model_inputs()
+    sparse_feature_columns = create_sparse_feature_columns()
+    features = layers.DenseFeatures(sparse_feature_columns)(inputs)
+
+    for units in hidden_units:
+        features = layers.Dense(units)(features)
+        features = layers.BatchNormalization()(features)
+        features = layers.ReLU()(features)
+        features = layers.Dropout(dropout_rate)(features)
+
+    outputs = layers.Dense(units=NUM_CLASSES, activation="softmax")(features)
+    model = keras.Model(inputs=inputs, outputs=outputs)
+    return model
+
+
+baseline_model = create_baseline_model()
+keras.utils.plot_model(baseline_model, show_shapes=True)
+
+run_experiment(baseline_model)
+
+"""
+The baseline linear model achieves ~76.4% test accuracy.
+"""
+
+"""
+## Experiment 2: wide and deep model
+
+In the second experiment, we create a wide and deep model. The wide part of the model a linear model,
+while the deep part of the model is a multi-layer feed-forward network.
+
+In the wide part of the model, we use the sparse representation of the input features, while we use the
+dense repesentation of the input features for the deep part of the model.
```

Comment:
Suggestions: 
- Refactor the sentence to avoid this repetition: "...we use..., while we use.."
- Fix typo ("representation"  ~~"repesentation"~~)


```suggestion
Use the sparse representation of the input features in the wide part of the model and the
dense representation of the input features for the deep part of the model.
```